solr/test-framework/src/java/org/apache/solr/util/NotSecurePsuedoRandom.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.util;
+
+import java.security.SecureRandom;
+import java.security.SecureRandomSpi;
+import java.util.Random;
+
+/**
+ * A mocked up instance of SecureRandom that just uses {@link Random} under the covers.
+ * This is to prevent blocking issues that arise in platform default
+ * SecureRandom instances due to too many instances / not enough random entropy.
+ * Tests do not need secure SSL.
+ */
+public class NotSecurePsuedoRandom extends SecureRandom {
+  public static final SecureRandom INSTANCE = new NotSecurePsuedoRandom();
+  private static final Random RAND = new Random(42);
```

Comment:
This must *not* be static or shared across test instances. A better solution would be to create this off an initial long seed and this seed should be taken from RandomizedContext.current..().random().nextLong().

Reply:
Hmm, not sure how that would work. Not every test runs with RandomizedContext, and even for those that do according to the code "static test class initializers are not permitted to access random contexts"

Reply:
If the concern is that the random seeds are not reproducible across test runs... nothing using SecureRandom right now is reproducible anyway by design, so this wouldn't be a regression.

Reply:
All tests should extend from LuceneTestCase. If they don't, it's likely a problem already (because this also dodges thread leak checks, etc.). As for static variables - don't make it a static instance (it wasn't before if I see the diff right). Static instances make the reasoning about tests a lot more difficult and order-dependent.

bq. "static test class initializers are not permitted to access random contexts"

Correct. Because this happens outside of the control of the test runner. Again - it makes reasoning about leaked threads (and anything else) more difficult.